tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs

```diff
@@ -3611,6 +3611,9 @@ private MongoCollection<BsonDocument> GetCollection(MongoServer server)
 
         private MongoServerInstance GetPrimary(MongoServer server)
         {
+            SpinWait.SpinUntil(
```

Comment:
why is this change required?

Reply:
I think it's because `TestFindAndRemoveWithMaxTime` spins up a new `MongoClient` by passing in `retryWrites: false`, resulting in a race condition where we try to get the primary before the topology has finished initializing. 

See: https://jira.mongodb.org/browse/CSHARP-2755 and its PR for a bit more detail
(Robert's also looking more deeply into it).

Reply:
This change has been removed and replaced with Robert's "Equals" fix